engine/src/commands/commands.cc

```diff
@@ -725,28 +725,33 @@ int cmd_process_host_check_result(int cmd, time_t check_time, char* args) {
     return ERROR;
 
   // Get the host name.
-  char* host_name(args);
+  auto split = absl::StrSplit(args, ';');
+  auto split_it = split.begin();
+
+  if (split_it == split.end())
+    return ERROR;
 
   // Get the host check return code and output.
-  char* delimiter(strchr(host_name, ';'));
-  if (!delimiter)
-    return ERROR;
-  *delimiter = '\0';
-  ++delimiter;
-  char* output(strchr(delimiter, ';'));
-  if (output) {
-    *output = '\0';
-    ++output;
-  } else
-    output = "";
-  int return_code(strtol(delimiter, nullptr, 0));
+  std::string host_name = std::string(*split_it);
 
-  // replace \\n with \n
-  string::unescape(output);
+  int return_code;
+  ++split_it;
```

Comment:
split_it can be split.end(), isn't it?
So you have to check this

Reply:
yes,i will fix.